### DISCUSSION TITLE
Question regarding interface kernels

### DISCUSSION THREAD

**Discussion by souravmat-git** [idaholab/moose#22680] [Q&A General] [answered]
Hello, 

I have a question regarding interface kernels. I am trying to implement the interfacial conditions, Eqs. (3) and (4), in MOOSE (please refer to the following schematic):

![Schematic](https://user-images.githubusercontent.com/29055022/201437338-3e25cbe7-6a13-46b3-b49f-1096e21bec41.png)

Here $\phi^{\theta}$, $\sigma^{\theta}$, and $i^{\theta}$ refer to the electric potential, conductivity and current in the bulk domains $\theta = \alpha, \beta$. Eq. (3) ensures the current continuity, and Eq. (4) gives the interfacial current.

Further, I think these conditions are similar to the [ElectrostaticContactCondition](https://mooseframework.inl.gov/source/interfacekernels/ElectrostaticContactCondition.html), which is already in MOOSE. I found an example input file, [contact_conductance_calculated.i](https://github.com/idaholab/moose/blob/next/modules/electromagnetics/test/tests/interfacekernels/electrostatic_contact/contact_conductance_calculated.i), based on this kernel. 

However, my question is regarding the part of the code that ensures these conditions. Specifically,
```
switch (type)
  {
    case Moose::Element:
      res = -contact_conductance * (_neighbor_value[_qp] - _u[_qp]) * _test[_i][_qp];
      break;

    case Moose::Neighbor:
      res = _conductivity_primary[_qp] * _grad_u[_qp] * _normals[_qp] * _test_neighbor[_i][_qp];
      break;
  }
```
I am unsure how this code ensures current continuity, i.e. Eq. (3). 

I found a similar GitHub discussion concerning this [Interface kernel with coupled variable](https://github.com/idaholab/moose/discussions/22366#discussion-4463907). In this case, however, the residuals are implemented differently despite having similar interfacial conditions.

Are these implementations equivalent, or am I missing something?

I welcome clarifications/suggestions regarding this.

Thanks,
Sourav

**Comment by hugary1995** [comment, marked as answer]
Yes, this ensures current continuity weakly.

**Reply by souravmat-git** [reply]
Thanks!
